app/controllers/nonsipp/membercontributions/MemberContributionListController.scala

```diff
@@ -26,74 +27,138 @@ import eu.timepit.refined.api.Refined
-      Redirect(
-        navigator.nextPage(ReportMemberContributionListPage(srn), mode, request.userAnswers)
-      )
-    } else {
+      if (memberListSize > Constants.maxSchemeMembers) {
+        Future.successful(
+          Redirect(
+            navigator.nextPage(MemberContributionsListPage(srn), mode, request.userAnswers)
+          )
+        )
+      } else {
 
-      form
-        .bindFromRequest()
-        .fold(
-          errors =>
-            BadRequest(
-              view(errors, ReportMemberContributionListController.viewModel(srn, page, mode, memberList, userAnswers))
-            ),
-          _ =>
-            Redirect(
-              navigator
-                .nextPage(ReportMemberContributionListPage(srn), mode, request.userAnswers)
-            )
+        form
+          .bindFromRequest()
+          .fold(
+            errors =>
+              Future.successful(
+                BadRequest(
+                  view(
+                    errors,
+                    MemberContributionListController.viewModel(srn, page, mode, memberList, userAnswers)
+                  )
+                )
+              ),
+            value =>
+              for {
+                updatedUserAnswers <- buildUserAnswerBySelection(srn, value, memberListSize)
+                _ <- saveService.save(updatedUserAnswers)
+                submissionResult <- if (value) {
+                  psrSubmissionService.submitPsrDetails(srn)(
+                    implicitly,
+                    implicitly,
+                    request = DataRequest(request.request, updatedUserAnswers)
+                  )
+                } else {
+                  Future.successful(Some(()))
+                }
+              } yield submissionResult.getOrRecoverJourney(
+                _ =>
+                  Redirect(
+                    navigator
+                      .nextPage(EmployerContributionsMemberListPage(srn), mode, request.userAnswers)
+                  )
+              )
+          )
+      }
+  }
+
+  private def buildUserAnswerBySelection(srn: Srn, selection: Boolean, memberListSize: Int)(
+    implicit request: DataRequest[_]
+  ): Future[UserAnswers] = {
+    val userAnswerWithMemberContList = request.userAnswers.set(MemberContributionsListPage(srn), selection)
+
+    if (selection) {
+      val indexes = (1 to memberListSize)
+        .map(i => refineV[OneTo300](i).leftMap(new Exception(_)).toTry)
+        .toList
+        .sequence
+
+      Future.fromTry(
+        indexes.fold(
+          _ => userAnswerWithMemberContList,
+          index =>
+            index.foldLeft(userAnswerWithMemberContList) {
+              case (uaTry, index) =>
+                val optTotalMemberContribution = request.userAnswers.get(TotalMemberContributionPage(srn, index))
+                for {
+                  ua <- uaTry
+                  ua1 <- ua.set(TotalMemberContributionPage(srn, index), optTotalMemberContribution.getOrElse(Money(0)))
```

Comment:
I don't think we should set anything to 0 in the FE.
Let's have a chat about this @cyeniceri

Reply:
its setting to 0 by default even we dont report any member contributions

Reply:
We had a chat and we'll discuss alternative solutions on the sprint review.